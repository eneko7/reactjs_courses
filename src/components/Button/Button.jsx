import React from 'react';
import PropTypes from 'prop-types';
import styles from './Button.scss';

const Button = ({ text, onClick }) => (
  <button type="button" className={styles.header_button} onClick={onClick}>{text}</button>
);

Button.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Button;
