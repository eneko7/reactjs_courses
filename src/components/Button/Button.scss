@import "../../assets/constants/constants.scss";

.header_button {
  cursor: pointer;
  color: #fff;
  border: none;
  background-color: $roseColor;
  padding: 5px 100px;
  margin-top: 15px;
  font-size: 16px;
  font-family: $fontFamilyBold;
  text-transform: uppercase;
}

.header_button:focus {
  outline: none;
}

@media (max-width: $tablet) {
  .header_button {
    margin: 15px auto;
    display: block;
    padding: 7px 100px;
  }
}
