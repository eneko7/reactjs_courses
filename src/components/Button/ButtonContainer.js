import { connect } from 'react-redux';
import Button from './Button';
import { getSearchBy, getSortBy, getSearchRequest } from '../../modules/filter/filterSelectors';
import { fetchSearchMovies } from '../../modules/movies/moviesActions';

const mapStateToProps = state => ({
  searchRequest: getSearchRequest(state),
  sortBy: getSortBy(state),
  searchBy: getSearchBy(state),
});

const mapDispatchToProps = {
  fetchSearchMovies,
};

export default connect(mapStateToProps, mapDispatchToProps)(Button);
