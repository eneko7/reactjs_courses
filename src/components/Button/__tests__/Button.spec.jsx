import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Button from '../Button';

const renderer = new ShallowRenderer();
const data = {
  text: 'text',
};
renderer.render(<Button {...data} />);
describe('Button', () => {
  it('renders correctly', () => {
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
