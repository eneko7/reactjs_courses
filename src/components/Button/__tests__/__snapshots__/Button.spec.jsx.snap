// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="header_button"
  onClick={[Function]}
  type="button"
>
  text
</button>
`;
